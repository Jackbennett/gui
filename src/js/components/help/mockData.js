export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
