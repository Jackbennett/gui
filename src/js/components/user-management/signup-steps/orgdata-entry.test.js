import React from 'react';
import renderer from 'react-test-renderer';
import { undefineds } from '../../../../../tests/mockData';
import OrgDataEntry from './orgdata-entry';

describe('Login Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<OrgDataEntry setSnackbar={jest.fn} onSubmit={jest.fn} recaptchaSiteKey="test" />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
