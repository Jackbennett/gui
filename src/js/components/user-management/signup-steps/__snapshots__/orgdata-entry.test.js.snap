// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
<div>
  <h1>
    Setting up your Account
  </h1>
  <h2
    class="margin-bottom-large"
  >
    To finish creating your account,
    <br />
    please fill in a few details
  </h2>
  <form
    class=""
  >
    <div
      class="MuiFormControl-root  required"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="name"
      >
        Company or organization name *
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        required=""
        style="width: 400px; max-width: 100%;"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="name"
          name="name"
          placeholder="Company or organization name *"
          required=""
          type="text"
          value=""
        />
      </div>
      <p
        class="MuiFormHelperText-root"
        id="component-error-text"
      />
    </div>
    <div
      class="MuiFormControl-root  required"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="email"
      >
        Email *
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        required=""
        style="width: 400px; max-width: 100%;"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="email"
          name="email"
          placeholder="Email *"
          required=""
          type="text"
          value=""
        />
      </div>
      <p
        class="MuiFormHelperText-root"
        id="component-error-text"
      />
    </div>
    <label
      class="MuiFormControlLabel-root"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-4"
            data-indeterminate="false"
            id="tos"
            name="tos"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        <label
          for="tos"
        >
          By checking this you agree to our
           
          <a
            href="https://northern.tech/legal/Hosted%20Mender%20Agreement%20-%2005-23-2020%20-%20Northern.tech%20AS.pdf"
            rel="noopener noreferrer"
            target="_blank"
          >
            Terms of service
          </a>
           
          and
           
          <a
            href="https://northern.tech/legal/privacy-policy"
            rel="noopener noreferrer"
            target="_blank"
          >
            Privacy Policy
          </a>
           
          *
        </label>
      </span>
    </label>
    <label
      class="MuiFormControlLabel-root"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-4"
            data-indeterminate="false"
            id="marketing"
            name="marketing"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        By checking this you agree that we can send you occasional email updates about Mender. You can unsubscribe from these emails at any time
      </span>
    </label>
    <div
      class="margin-top"
      validations=""
    >
      <div />
    </div>
    <div
      class="flexbox"
      style="justify-content: flex-end; margin-top: 32px;"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled=""
        id="login_button"
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Complete signup
        </span>
      </button>
    </div>
  </form>
</div>
`;
